src/tests/lxc-test-unpriv

```diff
@@ -39,8 +39,8 @@ if modprobe -q overlayfs; then
 
         mount -t tmpfs none ${MOUNTDIR}
 
-        mkdir "${MOUNTDIR}/lowerdir" "${MOUNTDIR}/upperdir" "${MOUNTDIR}/overlayfs"
-        mount -t overlayfs -o lowerdir="${MOUNTDIR}/lowerdir",upperdir="${MOUNTDIR}/upperdir" none "${MOUNTDIR}/overlayfs"
+        mkdir ${MOUNTDIR}/{lowerdir,upperdir,workdir,overlayfs}
+        mount -t overlayfs -o lowerdir="${MOUNTDIR}/lowerdir",upperdir="${MOUNTDIR}/upperdir",workdir="${MOUNTDIR}/workdir" none "${MOUNTDIR}/overlayfs"
 
```

Comment:
Thanks for the patch. One nit though. What happens on older kernels whose overlay module does not support the workdir option? Should the logic be:
- try with workdir option first and if that fails try without the workdir option?

Reply:
Yeah, looks like this is going to break older kernels, just a reminder that LXC 2.0.x still supports kernels from 2.6.32 onwards so this does need fixing as the ultimate target for this fix is 2.0.x

Reply:
Hi, thanks for catching this. 

To this test itself, the script will skip this overlayfs test part on older kernels (tested on Trusty 3.13, which needs the overlayfs module) with an if statement:

    if modprobe -q overlayfs; then

So I think as long as this module check criteria is valid, this change should be fine?

Reply:
Ok, cool.